smack-extensions/src/main/java/org/jivesoftware/smackx/softwareinfo/SoftwareInfoManager.java

```diff
@@ -0,0 +1,144 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.softwareinfo;
+
+import java.io.IOException;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.WeakHashMap;
+
+import org.jivesoftware.smack.Manager;
+import org.jivesoftware.smack.SmackException.NoResponseException;
+import org.jivesoftware.smack.SmackException.NotConnectedException;
+import org.jivesoftware.smack.XMPPConnection;
+import org.jivesoftware.smack.XMPPException.XMPPErrorException;
+import org.jivesoftware.smack.parsing.SmackParsingException;
+import org.jivesoftware.smack.xml.XmlPullParserException;
+import org.jivesoftware.smackx.disco.ServiceDiscoveryManager;
+import org.jivesoftware.smackx.disco.packet.DiscoverInfo;
+import org.jivesoftware.smackx.disco.packet.DiscoverInfo.Identity;
+import org.jivesoftware.smackx.disco.packet.DiscoverInfoBuilder;
+import org.jivesoftware.smackx.softwareinfo.element.SoftwareInfoElement;
+
+import org.jxmpp.jid.Jid;
+
+/**
+* Entry point for Smack's API for XEP-0232: Software Information.
+* <br>
+* To add {@link SoftwareInfoElement}, use {@link #addSoftwareInfoElement(SoftwareInfoElement)} method.
+* <br>
+* In order to be updated with other user's Software Information add {@link SoftwareInfoListener}, use {@link #addSoftwareInfoListener(SoftwareInfoListener)}.
+* <br>
+* @see <a href="https://xmpp.org/extensions/xep-0232.html">
+*     XEP-0232 : Software Information.
+*     </a>
+*/
+public final class SoftwareInfoManager extends Manager {
+
+    private static final String FEATURE = "http://jabber.org/protocol/disco";
+    private static final Map<XMPPConnection, SoftwareInfoManager> INSTANCES = new WeakHashMap<>();
+    private final Set<SoftwareInfoListener> softwareInfoListeners = new HashSet<>();
+    private final ServiceDiscoveryManager serviceDiscoveryManager;
+    private DiscoverInfoBuilder discoverInfoBuilder;
+
+    public static synchronized SoftwareInfoManager getInstanceFor (XMPPConnection connection) throws IOException, XmlPullParserException, SmackParsingException {
+        SoftwareInfoManager manager = INSTANCES.get(connection);
+        if (manager == null) {
+            manager = new SoftwareInfoManager(connection);
+            INSTANCES.put(connection, manager);
+        }
+        return manager;
+    }
+
+    private SoftwareInfoManager(XMPPConnection connection) throws IOException, XmlPullParserException, SmackParsingException {
+        super(connection);
+        serviceDiscoveryManager = ServiceDiscoveryManager.getInstanceFor(connection);
+        addFeature();
```

Comment:
I *think* that adding the disco feature is unnecessary in this place, as the ServiceDiscoveryManager already should announce that by itself.